File: 2015/1/python/main.py
if __name__ == "__main__":
    with open("../input.txt", "r") as f:
        instructions = f.read()
        floor = 0
        entered_basement = False
        for i, c in enumerate(instructions):
            if c == '(':
                floor = floor + 1
            elif c == ')':
                floor = floor - 1
            if (floor == -1) and not entered_basement:
                entered_basement = True
                print(f"Basement entered at index {i}")
        print(f"Final floor: {floor}")

File: 2015/1/c/main.c
#include <math.h>
#include <stdio.h>

int main() {
				int floor = 0;
				char c;
				int count = 0;
				int entered_basement = -1;

				FILE * file = fopen("../input.txt", "r");

				if (file == NULL) {
								printf("File couldn't be opened\n");
								return 1;
				}

				while ((c = fgetc(file)) != EOF)	{
								switch (c) {
												case '(':
																++floor;
																break;
												case ')':
																--floor;
																break;
												default:
																break;
								}
								++count;
								if (entered_basement == -1 && floor < 0) entered_basement = count;
				}
				printf("First time in basement: %d\nFinal floor: %d\n", floor);
				return 0;
}	
